subsys/mpsl/cx/thread/mpsl_cx_thread.c

```diff
@@ -102,6 +104,7 @@ static void gpiote_irq_handler(const struct device *gpiob, struct gpio_callback
 			last_notified = granted_ops;
 			callback(granted_ops);
 		}
+		is_handled--;
```

Comment:
Let's imagine `register_callback()` is executing this function: it calls `callback(0)`, and just before this instruction GPIOTE IRQ is triggered. GPIOTE IRQ handler would check `if (callback != NULL && is_handled == 0)` what would evaluate to `false`, because `is_handled` would be `1`. GPIOTE IRQ handler would not call the callback. Then execution would return to `register_callback()` context which would decrement `is_handled` value, also not calling the callback. We would lose a callback which should be called because of GPIO state change.